python/hail/api2/matrixtable.py

```diff
@@ -2031,6 +2031,42 @@ def filter_cols_list(self, keys, keep=True):
             jkeys.append(ck_type._convert_to_j(k))
         return MatrixTable(self._jvds.filterSamplesList(jkeys, keep))
 
+    @handle_py4j
+    @typecheck_method(mapping=listof(strlike))
+    def reorder_columns(self, mapping):
+        """Reorder columns.
+
+        Examples
+        --------
+
+        Randomly shuffle order of samples:
```

Comment:
the method is called `reorder_columns` but then we say we are reordering samples. I think we should use `column` and `column ID` throughout even though matrix_table still has sample_ids as a fields. That should change to column_ids as generic-alization continues, though not in this PR